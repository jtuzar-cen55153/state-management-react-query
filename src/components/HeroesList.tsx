import { FC, ReactNode } from 'react';
import { ListGroup } from 'reactstrap';
import { Hero } from '../interface/hero';

export const HeroesList: FC<{ horizontal?: boolean; items?: Hero[]; render: (item: Hero) => ReactNode }> = ({
  horizontal = false,
  items,
  render,
}) =>
  items && items.length > 0 ? (
    <ListGroup horizontal={horizontal} data-testid="heroes-list">
      {items?.map(item => render(item))}
    </ListGroup>
  ) : null;
