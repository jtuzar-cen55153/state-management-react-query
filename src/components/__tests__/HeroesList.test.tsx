import { render } from '@testing-library/react';
import { HeroesList } from '../HeroesList';

const mockHeroes = [
  { id: '1', name: 'Hero 1' },
  { id: '2', name: 'Hero 2' },
  { id: '3', name: 'Hero 3' },
];

describe('<HeroesList />', () => {
  it('should render null', () => {
    const { container } = render(
      <HeroesList horizontal items={[]} render={item => <div key={item.id}>{item.name}</div>} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render for horizontal layout', () => {
    const { container } = render(
      <HeroesList horizontal items={mockHeroes} render={item => <div key={item.id}>{item.name}</div>} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render for vertical layout', () => {
    const { container } = render(
      <HeroesList items={mockHeroes} render={item => <div key={item.id}>{item.name}</div>} />,
    );

    expect(container).toMatchSnapshot();
  });
});
