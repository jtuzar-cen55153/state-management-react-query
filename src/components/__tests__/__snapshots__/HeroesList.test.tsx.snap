// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroesList /> should render for horizontal layout 1`] = `
<div>
  <ul
    class="list-group list-group-horizontal"
    data-testid="heroes-list"
  >
    <div>
      Hero 1
    </div>
    <div>
      Hero 2
    </div>
    <div>
      Hero 3
    </div>
  </ul>
</div>
`;

exports[`<HeroesList /> should render for vertical layout 1`] = `
<div>
  <ul
    class="list-group"
    data-testid="heroes-list"
  >
    <div>
      Hero 1
    </div>
    <div>
      Hero 2
    </div>
    <div>
      Hero 3
    </div>
  </ul>
</div>
`;

exports[`<HeroesList /> should render null 1`] = `<div />`;
